src/handler/sign_psbt.c

```diff
@@ -1788,11 +1888,22 @@ static void sign_segwit_v1(dispatcher_context_t *dc) {
         }
         crypto_hash_update(&sighash_context.header, tmp, 4);
 
+        if (43 != call_get_merkleized_map_value(dc,
+                                                &state->cur.in_out.map,
+                                                (uint8_t[]){PSBT_IN_WITNESS_UTXO},
+                                                1,
+                                                tmp,
+                                                43)) {
```

Comment:
I'm a bit uneasy with the assumption here: the PSBT_IN_WITNESS_UTXO is `<64-bit int> <scriptPubKeylen> <scriptPubKey>`, whose length is `8 + 1 + <length of scriptPubKey>` (well, assuming that the scriptPubKey is not bigger than 252 bytes, which should be a safe assumption). We don't currently support anything with <length of scriptPubKey> > 32, but it could actually be smaller (e.g. a P2SH's scriptPubkey is 22 bytes, because it uses a 20-byte hash, so this check would fail for nested segwit, see [here](bip-0141.mediawiki#p2wpkh-nested-in-bip16-p2sh) − not the case here since we are in segwitv1).

P2TR is currently the only transaction type supported in Taproot, so the assumption currently holds.

Since we only use the first 8 bytes, just checking that it's large enough should be more robust:
```
        if (8 > call_get_merkleized_map_value(dc,
                                              &state->cur.in_out.map,
                                              (uint8_t[]){PSBT_IN_WITNESS_UTXO},
                                              1,
                                              tmp,
                                              8 + 1 + MAX_PREVOUT_SCRIPTPUBKEY)) {
```

Note I also change 43 to the formula `8 + 1 + MAX_PREVOUT_SCRIPTPUBKEY`, so it generalizes if in the future we support larger scriptPubKeys by just changing `MAX_PREVOUT_SCRIPTPUBKEY`.

Reply:
I refered to the value specified in the BIP341 for this as you can see [here]. But I can definitely change it.(https://bips.xyz/341#:~:text=amount%20(8)%3A%20value,always%2035%20bytes.)

Reply:
You're right; I thought leaving `state->cur.input.prevout_amount` would work, but it is indeed not initialized when processing outputs (unlike when processing inputs!), so it was wrong before.